### ISSUE TITLE
NTR: HuBMAP Move FMA terms to Uberon Vasculature, heart, lung

### ISSUE THREAD

**Issue by emquardokus** [obophenotype/uberon#2035]
[FMA _term_List.csv](https://github.com/obophenotype/uberon/files/7043104/FMA._term_List.csv)
**Google sheet here of same terms as in FMA._term_List.csv used to track progress:** https://docs.google.com/spreadsheets/d/1ECi6VqDIcB63lJAQtipN-bW4WOhUuawXOqnxX8LMnqo/edit?usp=sharing

**Preferred term label:**

**Synonyms**

**Definition (free text, please give PubMed ID)**

**Parent term (use https://www.ebi.ac.uk/ols/ontologies/uberon)**

**Your nano-attribution (ORCID)**
https://orcid.org/0000-0001-7655-4833

**Comment by emquardokus** [comment]
@dosumis 
This ticket is related to Import vasculature connectivity relationships from VCCF (Griffin Weber) #1681
https://github.com/obophenotype/uberon/issues/1681
In addition to the vasculature FMA terms, there are heart and lung also needed for HuBMAP ASCT+B tables

**Comment by cmungall** [comment]
We already have 17 of these

[Term]
id: UBERON:0039211
name: jejunal artery
xref: FMA:14808 ! jejunal artery

[Term]
id: UBERON:0039212
name: right colic artery
xref: FMA:14811 ! right colic artery

[Term]
id: UBERON:0039219
name: left colic artery
xref: FMA:14826 ! left colic artery

[Term]
id: UBERON:0039227
name: left colic vein
xref: FMA:15394 ! left colic vein

[Term]
id: UBERON:0039231
name: right colic vein
xref: FMA:15407 ! right colic vein

[Term]
id: UBERON:0039244
name: deep artery of penis
xref: FMA:19794 ! deep artery of penis

[Term]
id: UBERON:0039263
name: saphenous branch of descending genicular artery
xref: FMA:22510 ! saphenous branch of descending genicular artery

[Term]
id: UBERON:0039304
name: circumflex branch of left coronary artery
xref: FMA:3895 ! circumflex branch of left coronary artery

[Term]
id: UBERON:0039341
name: dorsal venous arch of foot
xref: FMA:44356 ! dorsal venous arch of foot

[Term]
id: UBERON:0039351
name: dorsal digital artery of foot
xref: FMA:44660 ! dorsal digital artery of foot

[Term]
id: UBERON:0039393
name: infrahyoid artery
xref: FMA:49474 ! infrahyoid artery

[Term]
id: UBERON:0039452
name: straight sinus
xref: FMA:50769 ! straight sinus

[Term]
id: UBERON:0039545
name: right superior lobar bronchus
xref: FMA:7397 ! right superior lobar bronchus

[Term]
id: UBERON:0039548
name: middle lobar bronchus
xref: FMA:7401 ! middle lobar bronchus

[Term]
id: UBERON:0039556
name: left superior lobar bronchus
xref: FMA:7423 ! left superior lobar bronchus

[Term]
id: UBERON:0039562
name: left inferior lobar bronchus
xref: FMA:7432 ! left inferior lobar bronchus

[Term]
id: UBERON:0039600
name: myocardium of right ventricle
xref: FMA:9535 ! myocardium of right ventricle

**Comment by cmungall** [comment]
This request is actually quite tricky. We also need to bring in some of the closure of requested terms

e.g. to bring in right first lumbar artery (in the list) we need the generic First lumbar artery (not on the list)

Historically we have limited minting IDs for serially or laterally repeated entities where the biology is identical. We should be dynamic and respond to hubmap requests. But this may be quite challenging with current resources, and doing this in a piecemeal way could lead to significant technical debt

**Comment by emquardokus** [comment]
@cmungall  worth a bit of discussion; I am willing to work through the issues if I know what all of them are. Each step forward is worth a lot.  The list provided was obviously just what we could automatically generate from going through all of our ASCT+B tables as in FMA and at the time not in Uberon (I'll update the ones you point out above that are now in), so now the rolling up sleeves and filling in blanks is needed. Thank you for examples. 
I'm working on the self guided training that Nico M. and Nicole V. have been providing, so I should be able to be more highly functional as to the do's and don'ts via Uberon and CL soon.

**Comment by cmungall** [comment]
Made a start on the more straightforward terms, and on some terms that will be needed to link ones in the list:

#2042

**Comment by cmungall** [comment]
Thanks @emquardokus 

I made a new issue for anticipating general requirements and for discussing some general high level design issues from this issue: #2043.  

Let's use this issue to continue honing in on this one request

**Comment by paolaroncaglia** [comment]
@emquardokus 
Shall we transfer your Excel spreadsheet to a Google one, so we can update it more easily (with the 17 existing terms in https://github.com/obophenotype/uberon/issues/2035#issuecomment-910604514 + the ones that Chris created in https://github.com/obophenotype/uberon/pull/2042/commits/e7aa4441aeabafff31c4ea335a6eca1466082a28). Then we could use the same spreadsheet to split tasks and record progress - I'm happy to help - I've added #2043 to the agenda for the next Uberon editors' meeting if we need to discuss the more complex terms.

**Comment by emquardokus** [comment]
@paolaroncaglia yes, I can put into a google sheet; I had already added the ones Chris mentioned to it so I could keep track on my own sheet.  How much of the field codes should I leave in; i.e. is there a google sheet template I should use for this?  I linearized so 1 row would be original FMAID, FMA label, the the new uberon ID assigned, uberon label and the fma xRef.  Thanks

**Comment by paolaroncaglia** [comment]
Hi @emquardokus ,

> @paolaroncaglia yes, I can put into a google sheet; I had already added the ones Chris mentioned to it so I could keep track on my own sheet. How much of the field codes should I leave in; i.e. is there a google sheet template I should use for this? I linearized so 1 row would be original FMAID, FMA label, the the new uberon ID assigned, uberon label and the fma xRef. Thanks

Thank you for moving to [this Google sheet](https://docs.google.com/spreadsheets/d/1ECi6VqDIcB63lJAQtipN-bW4WOhUuawXOqnxX8LMnqo/edit#gid=0), I confirm I can access it. There's no specific suggested template as far as I know (though that may change when discussion progresses, i.e. if we find ways to automate creation of some of those terms, but for now what you have is great!). We can discuss specific columns' content via Slack, I'll get to that later today. Thanks!

Paola

**Comment by paolaroncaglia** [comment]
As discussed with @dosumis and @emquardokus yesterday: we should definitely add new terms where left and right instances are asymmetrical in a species of interest, e.g. human lungs where one has 2 lobes and the other has 3. To be fair, Uberon already has 'left lung' and 'right lung'. If there are non-asymmetrical pairs in Ellen's list of missing FMA terms (morphologically and/or functionally), these could be prioritized for addition, possibly with a short rdfs:comment or definition gloss to describe the asymmetry.

**Comment by paolaroncaglia** [comment]
Copying minutes from yesterday's Uberon editors call:

- Generic: 

- - Walk up a tree to common term in & use that for parentage in Uberon

- The challenge - each branch presents its own issue.  Vasculature is particularly hard because of the representation of vessels/tree structures.
- Vasculature modelling issues (needs a mini-workshop)

- - Are vessels trees or does every segment get a name, or something in between.
- - What is/are the relationship(s) between connected vessels 
- - What is/are the relationship(s) between vasculature and tissues
- - Current high level sketch for approach https://github.com/obophenotype/uberon/wiki/Branching-structures-Design-Pattern 
-

**Comment by paolaroncaglia** [comment]
@emquardokus @dosumis 
I color-coded and sorted Ellen's spreadsheet by anatomical system, then I arranged the FMA labels in alphabetical order, to highlight the "pair types" and the terms that aren't part of a pair or set. Results [here](https://docs.google.com/spreadsheets/d/1ECi6VqDIcB63lJAQtipN-bW4WOhUuawXOqnxX8LMnqo/edit#gid=1087172102). 
Light blue = lung parts (almost all are members of pairs or sets)
Light red = vascular parts (many are members of pairs or sets, but some aren't; could e.g. FMA:85528 'Artery of pterygoid canal' be post-composed in the ASCT-B table using the Uberon terms for artery and for 'vidian canal'? vidian canal has exact synonym pterygoid canal)
Bright red = heart parts (these could all be post-composed)
No fill = none of the above (2 terms only, I can add them if you'd like).

**Comment by paolaroncaglia** [comment]
Note that some vascular terms were added in the latest Uberon release (e.g. 'anterior intercostal artery'), so
- [ ] the table needs updating (which sheet should we update?)

See https://github.com/obophenotype/uberon/releases/tag/v2021-09-29 for the list of new terms.

**Comment by emquardokus** [comment]
@paolaroncaglia the link is above where I state I copied list to Google sheet like you asked me to do: : https://docs.google.com/spreadsheets/d/1ECi6VqDIcB63lJAQtipN-bW4WOhUuawXOqnxX8LMnqo/edit?usp=sharing thanks very much for progress report!!

**Comment by paolaroncaglia** [comment]
@dosumis cc @emquardokus 
I don't have anything to add/update wrt what I outlined in my previous comments. If you have any updates or notes from offline discussions, please note them here for me if I can't join today's meeting. Thank you!

**Comment by paolaroncaglia** [comment]
@dosumis let me know if you'd like me to help with anything from [my previous comment](https://github.com/obophenotype/uberon/issues/2035#issuecomment-951659040).

**Comment by paolaroncaglia** [comment]
Update: 
As fare as I'm aware, the missing terms described [here](https://github.com/obophenotype/uberon/issues/2035#issuecomment-951659040) are still missing, except for 17 new terms that Nico added, listed [here](https://github.com/obophenotype/uberon/commit/49154a41a5e976bb6fd2860fbcbb205e4473719b).  
It's been some time, so the original list may need reviewing to know exactly what terms are still missing.

**Comment by emquardokus** [comment]
> We already have 17 of these
> 
> [Term] id: UBERON:0039211 name: jejunal artery xref: FMA:14808 ! jejunal artery
> 
> [Term] id: UBERON:0039212 name: right colic artery xref: FMA:14811 ! right colic artery
> 
> [Term] id: UBERON:0039219 name: left colic artery xref: FMA:14826 ! left colic artery
> 
> [Term] id: UBERON:0039227 name: left colic vein xref: FMA:15394 ! left colic vein
> 
> [Term] id: UBERON:0039231 name: right colic vein xref: FMA:15407 ! right colic vein
> 
> [Term] id: UBERON:0039244 name: deep artery of penis xref: FMA:19794 ! deep artery of penis
> 
> [Term] id: UBERON:0039263 name: saphenous branch of descending genicular artery xref: FMA:22510 ! saphenous branch of descending genicular artery
> 
> [Term] id: UBERON:0039304 name: circumflex branch of left coronary artery xref: FMA:3895 ! circumflex branch of left coronary artery
> 
> [Term] id: UBERON:0039341 name: dorsal venous arch of foot xref: FMA:44356 ! dorsal venous arch of foot
> 
> [Term] id: UBERON:0039351 name: dorsal digital artery of foot xref: FMA:44660 ! dorsal digital artery of foot
> 
> [Term] id: UBERON:0039393 name: infrahyoid artery xref: FMA:49474 ! infrahyoid artery
> 
> [Term] id: UBERON:0039452 name: straight sinus xref: FMA:50769 ! straight sinus
> 
> [Term] id: UBERON:0039545 name: right superior lobar bronchus xref: FMA:7397 ! right superior lobar bronchus
> 
> [Term] id: UBERON:0039548 name: middle lobar bronchus xref: FMA:7401 ! middle lobar bronchus
> 
> [Term] id: UBERON:0039556 name: left superior lobar bronchus xref: FMA:7423 ! left superior lobar bronchus
> 
> [Term] id: UBERON:0039562 name: left inferior lobar bronchus xref: FMA:7432 ! left inferior lobar bronchus
> 
> [Term] id: UBERON:0039600 name: myocardium of right ventricle xref: FMA:9535 ! myocardium of right ventricle

Many of these uberon IDs are not findable currently, not even under obsolete terms--any idea where they went? 
There seem to be new uberon IDs now instead.
<img width="765" alt="Screen Shot 2022-06-15 at 4 06 07 AM" src="https://user-images.githubusercontent.com/47257000/173775913-5ec56372-993b-4488-9cae-21e4f25e879f.png">

**Comment by anitacaron** [comment]
@emquardokus I think you can use the correct uberon id column instead of the uberon id-not findable column

**Comment by emquardokus** [comment]
@anitacaron yes, that's what we had to do, but I wanted to mention this because Chris had looked up the "not findable" ones months ago and at that time they WERE FINDABLE... so I'm curious what happened to them since they literally disappeared.  This is not normal. Usually if something has be "obsoleted" it still exists as something you can find in the obsoleted files.  I'm concerned that these just vanished.  So we are good and had to substitute in the new findable uberon Ids I listed.  This comment is more of an FYI: why did any uberon IDs disappear from existence that were findable at one point and are there others?????????

**Comment by dosumis** [comment]
It is possible for IDs to be unstable pre-release If created within a release cycle, but If we are losing IDs in public releases, this is very serious.  Should be caught during release process. We need to investigate and patch ASAP.

**Comment by emquardokus** [comment]
I wonder what source Chris was using when he pulled the original list he provided using ROBOT?

**Comment by matentzn** [comment]
I have pinged Chris on slack, but we should _never_ curate data off a comment in an issue :) @anitacaron has convinced me to ignore Chris comment. In any case, for you @emquardokus the important thing is to use the actual ids (correct ids). That wont change no matter what.

**Comment by cmungall** [comment]
Apologies for any confusion I caused in my comment from last year. I believe I was chaining together a series of legacy perl scripts when I made that list. You can ignore the IDs

**Comment by bvarner-ebi** [comment]
@matentzn, can we close this ticket?

**Comment by emquardokus** [comment]
@bvarner-ebi @matentzn @dosumis 
I do not believe the full issue has been resolved; can we wait until David is back >Aug 23rd to discuss whole topic? Thanks

**Comment by dosumis** [comment]
@emquadokus Can we just make a new ticket with a list if the terms that need to be added?  This one has gotten very confusing with the problems around non existent Uberon IDs.

**Comment by bvarner-ebi** [comment]
Echoing @dosumis's comment above, can a fresh ticket be created, @emquardokus? It is not clear to me what work is required.

To that effect, I will remove myself from this ticket, but feel free to assign a new ticket to me if one is created.
